model/include/model/builddirectory.h

```diff
@@ -0,0 +1,37 @@
+#ifndef CC_MODEL_BUILDDIRECTORY_H
+#define CC_MODEL_BUILDDIRECTORY_H
+
+#include <memory>
+
+#include <odb/core.hxx>
+
+#include <model/buildaction.h>
+#include <model/file.h>
+
+namespace cc
+{
+namespace model
+{
+
+struct BuildAction;
+
+#pragma db object
+struct BuildDirectory
```

Comment:
I'm not sure why this has to be a brand new table, foreign key, and all that jazz for the storage. `BuildAction` already contains `string command;`, for example. If `"directory"` is just another member of the entries in a compilation database JSON (just like `"command"`), then why isn't this reflected in the schema? Every build action **MUST** have a build directory (it just may or may not exist or point to anything meaningful).

This struct, as it is now, won't do uniqueing of build directories either, because it is the build directory that refers to **one** build action each. If we want a separate record to only store one directory ("string") exactly once, then it is the buildaction that has to refer into this relation's key.

Reply:
I wanted to keep backwards compatibility with existing databases, that's why I didn't want to change `BuildAction`'s schema. Is that not a priority? In that case you're right, storing it in `BuildAction` would make more sense.

Deduplicating the strings is not worth the hassle, in the CodeCompass SQLite database `BuildAction` only takes up 0.012% of the space.

Reply:
We can easily reparse projects, backward-compatibility with existing databases is not a high priority.